rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/20_filtering.yaml

```diff
@@ -130,3 +130,37 @@ setup:
   - is_true: metadata
   - is_true: routing_table
   - is_true: routing_nodes
+
+---
+"Filtering the cluster state by indices using wildcards should work in routing table and metadata":
+  - do:
+      index:
+        index: index1
+        type:  type
+        id:    testing_document
+        body:
+            "text" : "The quick brown fox is brown."
+
+  - do:
+      index:
+        index: index2
+        type:  type
+        id:    testing_document
+        body:
+            "text" : "The quick brown fox is brown."
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      cluster.state:
+        metric: [ routing_table, metadata ]
+        index: [ index* ]
+
+  - is_false: metadata.indices.testidx
+  - is_false: routing_table.indices.testidx
+
+  - is_true: metadata.indices.index1
+  - is_true: routing_table.indices.index1
+  - is_true: metadata.indices.index2
+  - is_true: routing_table.indices.index2
```

Comment:
That's a bigger discussion that might deserve its own issue. I don't have a strong opinion on this.